int16 e1
int16 e2
int16 e3
int16 e4
